### DISCUSSION TITLE
Not working Windows 22H2

### DISCUSSION THREAD

**Discussion by mongoled** [kangyu-california/PersistentWindows#232] [General]
Hi, as per documentation.

The execuatable launches with the OS through task schedular, can be seen in task manager, but does not prevent "Rapid HPD" from kicking in.

If the program is ran manually from the excuatable it works as expected.

I noted that the execeutable does not appear in the system tray, ive added the option in "system settings" to show in the "taskbar" but this has made no difference.

Tried task schedular with "run with highest prvilages", no difference

Tried task schedular as "system", no difference
 
What am I doing wrong ?

Thanks

**Comment by daskiiii** [comment]
I've noticed the program attempting to start, show up in my task bar, and then disappear without doing anything, as well. Here's the error code I got:

Faulting application name: PersistentWindows.exe, version: 5.40.8379.25233, time stamp: 0x639501a2
Faulting module name: KERNELBASE.dll, version: 10.0.19041.2913, time stamp: 0xa1c3e870
Exception code: 0xe0434352
Fault offset: 0x000000000002cf19
Faulting process id: 0x4140
Faulting application start time: 0x01d99645234864ae
Faulting application path: C:\Program Files\PersistentWindows-master\PersistentWindows.exe
Faulting module path: C:\Windows\System32\KERNELBASE.dll
Report Id: 9841b030-83dc-49b5-b260-34e2730583e2
Faulting package full name: 
Faulting package-relative application ID:

**Reply by kangyu-california** [reply]
older PW versions would crash on startup due to github webpage format change, please manually upgrade to 5.41

**Reply by daskiiii** [reply]
I updated manually shortly before I received this error, and have tested several variations of "-delay_start 10" with higher value, one to any effect.

**Reply by kangyu-california** [reply]
You may continue to use 5.40, just add "-check_upgrade=0" to command option to avoid crash. 
or you may try the [5.41 patch](https://github.com/kangyu-california/PersistentWindows/files/11391195/PersistentWindows5.41p224_4.zip) which resolved some platform compatibility issues.

**Reply by daskiiii** [reply]
that update worked, thank you!

**Comment by kangyu-california** [comment]
Please take some time to review "Installation" or "Known Issues" section on readme, especially the "-delay_start 10" command option, which works for me.

**Comment by mongoled** [comment]
Well I have tried with the new patch.
No change for me.
The task runs on startup, PersistentWindows.exe is running in "Details" section of "Task Manager", the scheduled task never stops running and no icon appears in the taskbar. 
I will try from a different Windows 10 install.
On the current OS I am running, the user folders main directories, i.e. doucments, downloads, desktop etc are not stored on the local machine but on a different machine (server). Could this be a possible cause? Looking at the notes in the readme, it does not seem this should be an issue

**Reply by mongoled** [reply]
Just tried from a different Windows 10 22H2 OS and its doing exactly the same from that OS as what i have described above.
Its on the same machine but a different drive

**Reply by kangyu-california** [reply]
By default, PW uses local appdata folder (c:/users/user/AppData/Local/PersistentWindows) to store persistent database.

It seems you have roaming user profile on network which may have conflict with PW default setting.

You may try [PersistentWindows5.42_p232.zip](https://github.com/kangyu-california/PersistentWindows/files/11672912/PersistentWindows5.42_p232.zip) to check if it works on roaming user profile setting.

If the patch does not work, you may try running PW with "-redirect_appdata" command option which tells PW to store database in the same directory as PersistentWindows.exe instead of local appdata folder.

**Reply by mongoled** [reply]
Sorry, my last post should have been more clear.
The second Windows OS I tried is not saving to a server. Its a "regular" Windows 10 22H2 install with no tweaks.

I will try the executable you have linked and will report back.

Thank you

**Reply by mongoled** [reply]
Just tried the new executable, in the "regular" Windows 22HS OS and no change.

I checked the event logs and see no warnings/errors with regards to the executable.

Is there somewhere I can look to see why the executable is not loading to task bar ??

Thanks

** EDIT **
I tried placing the folder in "Program Files x86" and "C:"

![Untitled](https://github.com/kangyu-california/PersistentWindows/assets/134506928/e888c543-66f2-4360-887d-64139160069f)

**Reply by kangyu-california** [reply]
The trigger of the task should be "When user login" instead of  "At system startup", otherwise PW would start too soon when user context is not ready yet.

I have wrote some script to create auto startup task for PW, in case you may want to give it a try 
[auto_start_task.zip](https://github.com/kangyu-california/PersistentWindows/files/11684226/auto_start_task.zip)

1. Delete the auto start task you manually created.
2. Unzip the content of auto_start_task.zip into existing PW installation dir
3. Launch a DOS command console with admin privilege, go to the PW installation dir, and run "auto_start_pw.bat"
4. Now press F5 key in task scheduler to refresh task list, and you will see a new task named "StartPersistentWindows..." be created with proper settings.
5. I have put "-delay_start 10" as default PW command option in the ps_auto_start_pw.ps1 file, feel free to modify the command option directly in task scheduler or modify the script, and you are all set.

**Reply by mongoled** [reply]
Hi again! There was me hoping it was just something I was missing/had done wrong.

So I set the trigger to "At log on", but the issue persists.

I will now take a look at the script.

Thanks for hanging in there for me :)

![Untitled2](https://github.com/kangyu-california/PersistentWindows/assets/134506928/fe18a975-6eb2-41dc-b359-0ecd16fd0c5a)

**Reply by mongoled** [reply]
OK, found out what I was doing wrong.

Initially I was setting "Run whether a user is logged on or not" from the "General" tab and not setting this from the "Triggers" tab as you kindly pointed out.

Once I set the "General" tab setting to "Run only when a user is logged on" along with the change to the "Trigger" it loads correctly now.

For those who are going to give the script a go. Here is what I did 

Opened a powershell window as admin

Changed the directory to the root location of where the files were downloaded.

Ran the .bat file using ./ 
i.e. ./auto_start_pw.bat

After the task was created I updated the directory where PersistentWindows.exe resides in the "Actions" tab

Thanks ever so much for helping me with this and for the work you have done with this solution

:)
